src/lib/Libattr/attr_fn_acl.c

```diff
@@ -667,6 +672,51 @@ hacl_match(const char *can, const char *master)
 
 /**
  * @brief
+ * 	group acl order match - match two strings when user is in group
+ *
+ * @param[in] can - Canidate string (first parameter) is a euser name string.
+ * @param[in] master -  Master string (2nd parameter) is an entry from a group acl.
+ *
+ * Strings match if can is a member of master.
+ *
+ * @return	int
+ * @retval	0	if strings match
+ * @retval	1	if not
+ *
+ */
+
+static int
+gacl_match(const char *can, const char *master)
+{
+	int i, ng = 0;
+	struct passwd *pw;
+	struct group *gr;
+	gid_t *groups = NULL;
+
+	if ((pw = getpwnam(can)) == NULL)
+		return(1);
+
+	if (getgrouplist(can, pw->pw_gid, NULL, &ng) < 0) {
+		if ((groups = (gid_t *) malloc(ng * sizeof (gid_t))) == NULL)
+			return(1);
+		getgrouplist(can, pw->pw_gid, groups, &ng);
+	}
+
+	for (i = 0; i < ng; i++)
+		if ((gr = getgrgid(groups[i])) != NULL)
+			if (!strcmp(gr->gr_name, master)) {
+				free(groups);
+				return (0);
+			}
+
+	if (groups != NULL)
```

Comment:
We've been trying to cleanup these conditional calls to free() and also ensuring pointers to heap memory are initialized properly. You've already initialized groups to NULL, so you are safe to simply call free() here without testing groups.